gpkit/nomials.py

```diff
@@ -558,4 +610,54 @@ def __bool__(self):
         return self.__nonzero__()
 
 
+class SignomialConstraint(Signomial):
+    """A constraint of the general form posynomial >= posynomial
+    Stored internally (exps, cs) as a single Signomial (self <= 0)
+    Usually initialized via operator overloading, e.g. cc = (y**2 >= 1 + x - y)
+    Additionally retains input format (lhs vs rhs) in self.left and self.right
+    Form is self.left >= self.right.
+    """
+    def __str__(self):
+        return str(self.left) + self.oper_s + str(self.right)
+
+    def __repr__(self):
+        return repr(self.left) + self.oper_s + repr(self.right)
+
+    def _latex(self, unused=None):
+        return self.left._latex() + self.oper_l + self.right._latex()
+
+    def __init__(self, left, right, oper_ge=True):
+        """Initialize a constraint of the form left >= right
+        (or left <= right, if oper_ge is False).
+
+        Arguments
+        ---------
+        left: Signomial
+        right: Signomial
+        oper_ge: bool
+            If true, form is left >= right; otherwise, left <= right.
+
+        Note: Constraints initialized via operator overloading always take
+              the form left >= right, e.g. (x <= y) becomes (y >= x).
+
+        Note: Unlike Constraints, SignomialConstraints have units.
+        """
+        left = Signomial(left)
+        right = Signomial(right)
+        pgt, plt = (left, right) if oper_ge else (right, left)
+
+        from . import SIGNOMIALS_ENABLED
+        if not SIGNOMIALS_ENABLED:
+            raise TypeError("Cannot initialize SignomialConstraint "
+                            "without SignomialsEnabled.")
+
+        p = plt - pgt
+
+        super(SignomialConstraint, self).__init__(p)
```

Comment:
True, and agreed it would be nice to eliminate.

Unfortunately we can't have `SignomialConstraint` and `PosynomialConstraint` inherit from a generic `Constraint`, because one needs to inherit from `Signomial` and the other from `Posynomial`, so we'd end up in Python multiple inheritance hell.

We could consider replacing `left` and `right` with `sgt` and `slt`, which would constrain the user's options in terms of pretty printing. I suspect we might still end up with some duplication though.

Another ugly option would be to check the `__class__` in `Signomial.__str__`, so that all the constraint printing code goes there.